spec/models/candidate_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe Candidate, type: :model do
+
+  it "has a valid factory" do
+    expect(build(:candidate)).to be_valid
+  end
+
+  # Lazily loaded to ensure it's only used when it's needed
+  let(:cand) { build(:candidate) }
+  let(:saved) { create(:candidate) }
+
+  describe "ActiveModel validations" do
+    # Basic validations
+    it { expect(cand).to validate_uniqueness_of(:profile_id).scoped_to(:post_id,:election_id)
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.<br>Space missing after comma.